From deaff5623e991083426f50883dd2872d51cb7765 Mon Sep 17 00:00:00 2001
From: amatig <amatig@gmail.com>
Date: Fri, 7 Oct 2011 11:28:17 +0200
Subject: [PATCH] patch makefile

---
 sipXproxy/src/Makefile.am |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/sipXproxy/src/Makefile.am b/sipXproxy/src/Makefile.am
index 4c4c409..048f5d9 100644
--- a/sipXproxy/src/Makefile.am
+++ b/sipXproxy/src/Makefile.am
@@ -54,6 +54,8 @@ libsipXproxy_la_SOURCES = \
 	RouteState.cpp \
 	SipXProxyCseObserver.h \
 	SipXProxyCseObserver.cpp \
+	SipXProxyCstaObserver.h \
+        SipXProxyCstaObserver.cpp \
    CallStateEventBuilder.h \
    CallStateEventBuilder.cpp \
    CallStateEventBuilder_XML.h \
-- 
1.7.4.4

